l1-contracts/src/UniFiAVSManager.sol

```diff
@@ -55,35 +63,63 @@ contract UniFiAVSManager is UniFiAVSManagerStorage, IUniFiAVSManager, UUPSUpgrad
         _;
     }
 
+    /**
+     * @dev Internal function to get AVS operator status via staticcall
+     * @param operator The address of the operator
+     */
+    function _getAvsOperatorStatus(address operator)
+        internal
+        view
+        returns (IAVSDirectory.OperatorAVSRegistrationStatus)
+    {
+        (bool success, bytes memory data) = address(AVS_DIRECTORY).staticcall(
+            abi.encodeWithSelector(
+                bytes4(
+                    keccak256("avsOperatorStatus(address,address)")
+                ),
+                address(this),
+                operator
+            )
+        );
+        if (!success) {
+            revert AVSOperatorStatusCallFailed();
+        }
+        return abi.decode(data, (IAVSDirectory.OperatorAVSRegistrationStatus));
+    }
+
     /**
      * @dev Modifier to check if the operator is registered in the AVS
      * @param operator The address of the operator
      */
     modifier registeredOperator(address operator) {
         if (
-            AVS_DIRECTORY.avsOperatorStatus(address(this), operator)
-                == IAVSDirectory.OperatorAVSRegistrationStatus.UNREGISTERED
+            _getAvsOperatorStatus(operator) ==
+            IAVSDirectory.OperatorAVSRegistrationStatus.UNREGISTERED
         ) {
             revert OperatorNotRegistered();
         }
         _;
     }
 
-    /**
-     * @custom:oz-upgrades-unsafe-allow constructor
-     */
     constructor(
         IEigenPodManager eigenPodManagerAddress,
         IDelegationManager eigenDelegationManagerAddress,
-        IAVSDirectory avsDirectoryAddress
+        IAVSDirectory avsDirectoryAddress,
+        IRewardsCoordinator rewardsCoordinatorAddress
     ) {
         EIGEN_POD_MANAGER = eigenPodManagerAddress;
```

Comment:
Lets add address(0) reverts, so that we don't get any info, low issues on the reports